api/errors/errors.go

```diff
@@ -298,22 +298,38 @@ func NewBlobstoreUnavailableError(cause error) BlobstoreUnavailableError {
 	}
 }
 
-type CFCreateError struct {
+type ResourceNotFoundError struct {
 	apiError
 }
 
-func NewCFCreateError(cause error) CFCreateError {
-	return CFCreateError{
+func NewResourceNotFoundError(cause error) ResourceNotFoundError {
+	return ResourceNotFoundError{
 		apiError: apiError{
 			cause:      cause,
-			title:      "CF-Create",
+			title:      "ResourceNotFound",
```

Comment:
There is a chance this causes some unexpected behavior since clients might assume `ResourceNotFound` is a `404` (based on its name and [this existing error](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/errors/v2.yml#L78)). I haven't evaluated this code, but a quick search shows the CLI specifically looking for it: https://github.com/search?q=repo%3Acloudfoundry%2Fcli%20ResourceNotFound&type=code

This might be one of the cases where it is worth making a unique error title to be safe.

Reply:
I think in general CF assumes a unique error `title` per `httpStatus`/`code`.

Reply:
Does it make sense to combine this with the `ResourceNotReady` error to avoid any special handling by the CLI? I was also thinking that we don't really need both errors since they are identical in how the handle the underlying error.

Reply:
Yeah, I think `ResourceNotReady` would work fine. 👍 
There may be a subtle distinction between something that could eventually become ready and something that will never become ready, but for these errors I don't think that matters.

Reply:
I've removed the `ResourceNotFound` error and just used the `ResourceNotReady` error everywhere.